src/asl.ts

```diff
@@ -1622,12 +1642,14 @@ export namespace ASL {
           isLiteralOrTypeOfExpr(expr.left) ||
           isLiteralOrTypeOfExpr(expr.right)
         ) {
-          const [literalOrTypeOf, val] =
-            isLiteralOrTypeOfExpr(expr.left) || isLiteralOrTypeOfExpr(expr.left)
-              ? [expr.left, expr.right]
-              : [expr.right, expr.left];
-
-          if (literalOrTypeOf.kind === "TypeOfExpr") {
+          // typeof x === "string" -> left: typeOf, right: literal
+          // "string" === typeof x -> left: literal, right: typeOf
+          // x === 1 -> left: identifier, right: literal
+          const [literalExpr, val] = isLiteralExpr(expr.left)
```

Comment:
what happened to the typeof logic?

Reply:
I switched the logic, you had `literalOrTypeOf`. It was kind of awkward because the expression in typeof is the typeof and the literal is the other side. For a literal, it is the opposite, the literal is the literal and the expression is the other side.

So I look for the literal, which is `string`, `number`, etc. With the opposite side being an expression or a `typeof`.

There was a failing use case (see the change to the typeof test) where `"string" === typeof x` didn't work. 

So now the logic is simple, 1) look for the literal, 2) decide if one side is a `typeof`, 3) process.